### ISSUE TITLE
HELM evaluation framework tries to generate with `temperature=0` which leads to `inf` tensors

### ISSUE THREAD

**Issue by aniketmaurya** [Lightning-AI/lit-gpt#401]
HELM evaluation framework tries to generate with `temperature=0` which leads to `inf` tensors error.

I don't have much idea how LLM will react to adding `1e-9` to the temperature value but this would avoid the error [here](https://github.com/Lightning-AI/lit-gpt/blob/6f9e7b0d9230d084b797be0ce47ba5b55aef1029/generate/base.py#L65C32-L65C38).

```python

        # forward
        logits = model(x, max_seq_length, input_pos)
        logits = logits[0, -1] / (temperature + 1e-9)
```

**Comment by carmocca** [comment]
Do you know why is that? It's very strange. Do you have any links?

**Comment by aniketmaurya** [comment]
I don't have any link, unfortunately! I tried printing the input request where the temperature was set to 0.

**Comment by Andrei-Aksionov** [comment]
Weird assumption: this framework relies on this additive in the denominator part and thus wants to use the lowest possible [temperature](https://github.com/Andrei-Aksionov/nanoGPTplus/blob/c2eedfe406bf7c8851c393825c09280a7e329242/src/model/gpt_language_model/gpt.py#L429-L437) so during sampling the logits with the highest probability will be outputted. 
Though not sure how useful this information is 🙃.

**Comment by aniketmaurya** [comment]
thanks for finding this @carmocca! It seems that neurips client doesn't have this condition to the raw request - https://github.com/stanford-crfm/helm/pull/1693/files#diff-a9de0c80d20516c09d5131d26c59dfdab923a2c7cfd2e09b8814e9bc4bd7521eR47

**Comment by aniketmaurya** [comment]
https://github.com/stanford-crfm/helm/pull/1693 will close this issue
